src/aics-image-viewer/components/App/index.tsx

```diff
@@ -420,12 +432,32 @@ const App: React.FC<AppProps> = (props) => {
 
     const channelNames = aimg.imageInfo.channelNames;
     const newChannelSettings = setChannelStateForNewImage(channelNames);
+
+    // order is important:
+    // we need to remove the old volume before triggering channels unloaded,
+    // which may cause calls on View3d to the old volume.
+    view3d.removeAllVolumes();
     setAllChannelsUnloaded(channelNames.length);
+    placeImageInViewer(aimg, newChannelSettings);
     channelRangesRef.current = new Array(channelNames.length).fill(undefined);
 
     const requiredLoadSpec = new LoadSpec();
     requiredLoadSpec.time = viewerState.current.time;
 
+    // make the currently enabled channels "required":
+    // find all enabled indices in newChannelSettings:
+    const requiredChannelsToLoad = newChannelSettings
+      ? newChannelSettings.map((channel, index) => (channel.volumeEnabled ? index : -1)).filter((index) => index >= 0)
+      : [];
+
+    // add mask channel to required channels?
+    const maskChannelName = getCurrentViewerChannelSettings()?.maskChannelName || "";
+    const maskChannelIndex = channelNames.indexOf(maskChannelName);
+    if (maskChannelIndex >= 0 && !requiredChannelsToLoad.includes(maskChannelIndex)) {
+      requiredChannelsToLoad.push(maskChannelIndex);
+    }
+    requiredLoadSpec.channels = requiredChannelsToLoad;
```

Comment:
nit/optional: maybe a silly thing to optimize, but I might put this in one more `if` statement to null-check `maskChannelName` rather than giving it a fallback value `... || "";`

Reply:
I agree, I'll do that